control_panel_api/tools.py

```diff
@@ -19,35 +50,22 @@ class UnsupportedToolException(APIException):
     default_code = 'unsupported_tool'
 
 
-class Tool(object):
-
-    def __init__(self, name):
-        if not name in SUPPORTED_TOOL_NAMES:
-            raise UnsupportedToolException
+class BaseTool(HelmToolDeployMixin, Auth0ClientConfigMixin):
+    name = None
```

Comment:
This somehow confuses me. Sorry I know it's a stupid one. 

`name` is a class variable as it's attached to the class (e.g. to the `RStudio`), but then it's accessed using `self.name`, as if it was an instance variable. 

Is it because Python somehow fallbacks to class variables when it doesn't find them at instance scope?